itests/org.openhab.core.thing.tests/src/main/java/org/openhab/core/thing/link/ItemChannelLinkAssignDefaultTagsOSGiTest.java

```diff
@@ -0,0 +1,201 @@
+                .formatted(TEST_ITEM_NAME, ItemChannelLinkRegistry.USE_TAGS);
+
+        modelRepository.addOrRefreshModel(TEST_MODEL_NAME, new ByteArrayInputStream(input.getBytes()));
+
+        Item item = itemRegistry.get(TEST_ITEM_NAME);
+        assertNotNull(item);
+        assertEquals(3, item.getTags().size());
+        assertTrue(item.getTags().contains("Measurement"));
+        assertTrue(item.getTags().contains("Temperature"));
+        assertTrue(item.getTags().contains("CustomTag"));
+    }
+
+    @Test
+    public void assertItemDidNotAssignSecondChannelDefaultTags() {
+        ThingTypeUID thingTypeUID = new ThingTypeUID("test", "test");
+        ThingUID thingUID = new ThingUID(thingTypeUID, "test");
+        Channel channel1 = ChannelBuilder.create(new ChannelUID(thingUID, "test1"))
+                .withDefaultTags(Set.of("Switch", "Power")).build();
+        Channel channel2 = ChannelBuilder.create(new ChannelUID(thingUID, "test2"))
+                .withDefaultTags(Set.of("Measurement", "Temperature")).build();
+        ThingBuilder thingBuilder = ThingBuilder.create(thingTypeUID, thingUID).withChannel(channel1)
+                .withChannel(channel2);
+        thingRegistry.add(thingBuilder.build());
+
+        String input = "String %s { channel=\"test:test:test:test1\" [ %s=true ], channel=\"test:test:test:test2\" [ %s=true ] }"
+                .formatted(TEST_ITEM_NAME, ItemChannelLinkRegistry.USE_TAGS, ItemChannelLinkRegistry.USE_TAGS);
+
+        modelRepository.addOrRefreshModel(TEST_MODEL_NAME, new ByteArrayInputStream(input.getBytes()));
+
+        Item item = itemRegistry.get(TEST_ITEM_NAME);
+        assertNotNull(item);
+        assertEquals(2, item.getTags().size());
+        assertTrue(item.getTags().contains("Switch"));
+        assertTrue(item.getTags().contains("Power"));
+    }
+
+    @Test
+    public void assertItemAssignedOwnTags() {
+        ThingTypeUID thingTypeUID = new ThingTypeUID("test", "test");
+        ThingUID thingUID = new ThingUID(thingTypeUID, "test");
+        Channel channel = ChannelBuilder.create(new ChannelUID(thingUID, "test")).withDefaultTags(Set.of("foo", "bar"))
+                .build();
+        ThingBuilder thingBuilder = ThingBuilder.create(thingTypeUID, thingUID).withChannel(channel);
+        thingRegistry.add(thingBuilder.build());
+
+        String input = "String %s [tag1, tag2] { channel=\"test:test:test:test\" [ %s=true ] }"
+                .formatted(TEST_ITEM_NAME, ItemChannelLinkRegistry.USE_TAGS);
+
+        modelRepository.addOrRefreshModel(TEST_MODEL_NAME, new ByteArrayInputStream(input.getBytes()));
+
+        Item item = itemRegistry.get(TEST_ITEM_NAME);
+        assertNotNull(item);
+        assertEquals(2, item.getTags().size());
+        assertTrue(item.getTags().contains("tag1"));
+        assertTrue(item.getTags().contains("tag2"));
+        assertTrue(item.getTags().contains("this should fail"));
```

Comment:
@lolodomo / @jimtng as you can see this test does NOT fail .. which is a proof the the build system is not actually running this test suite. Indeed by placing a similar failing test in other OSGi test classes in `org.openhab.core.thing.tests` we can easily see that **_none of the OSGI tests in the module are actually being run._**

Reply:
https://github.com/openhab/openhab-core/issues/4921